github-runner-manager/src/github_runner_manager/openstack_cloud/health_checks.py

```diff
@@ -48,8 +52,11 @@ def check_runner(openstack_cloud: OpenstackCloud, instance: OpenstackInstance) -
             "Health check failed due to unable to find keyfile for %s", instance.server_name
         )
         return False
-    except SSHError:
-        logger.exception("SSH Failed on %s, marking as unhealthy.")
+    except _SSHError:
+        logger.exception(
+            "Unable to get SSH connection for instance %s, marking as unhealthy.",
+            instance.server_name,
+        )
         return False
```

Comment:
This is confusing and should probably not return False, but instead raise the health check error. The problem is that this would result in the server not being purged, and this could be a risky change.